### PR TITLE
WEBEU2 - Web DB II Module Challenge - Rui Hildt 🇧🇪

### PR DESCRIPTION
**Pull request by ruihildt**

### CODE REVIEW

File: data/migrations/01-createCarsTable.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+exports.up = function(knex) {
+    return knex.schema.createTable('cars', table => {
+        table.increments();
+        table.text('vin').unique().notNullable();
+        table.text("make").notNullable();
+        table.text('model').notNullable();
+        table.integer('mileage');
```
Comment (delbaoliveira) [inline data/migrations/01-createCarsTable.js]: You can also add the `transmission type` and `status` and not add `.notNullable()`. That way if a new car is posted, the values will initially be null and can be edited later with a PUT request.

File: data/migrations/02-addTransmissionAndStatusToCarsTable.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+exports.up = function(knex) {
+  return knex.schema.alterTable('cars', table => {
```
Comment (delbaoliveira) [inline data/migrations/02-addTransmissionAndStatusToCarsTable.js]: This is a good alternative 🔥 But just be careful as `alterTable()` is not supported by SQLite according to the Knex Docs. More info: https://knexjs.org/#Schema-alter

Comment (ruihildt) [reply data/migrations/02-addTransmissionAndStatusToCarsTable.js]: As I understood the docs, it's `column.alter()` that's not supported by SQLite3.

File: data/migrations/01-createCarsTable.js (hunk 1 above)
Comment (ruihildt) [reply data/migrations/01-createCarsTable.js]: I used a second migration file with the `status` and `transmission` to simulate a functionality added afterwards. 
And I didn't use `notNullable()` on those. 

See https://github.com/ruihildt/webdb-ii-challenge/blob/master/data/migrations/02-addTransmissionAndStatusToCarsTable.js

File: data/migrations/02-addTransmissionAndStatusToCarsTable.js (hunk 2 above)
Comment (delbaoliveira) [reply data/migrations/02-addTransmissionAndStatusToCarsTable.js]: Hi Rui, 

Yes, it seems like I misread it. There are certain things that are allowed with alterTable in SQLite. Mainly allowing `user to rename a table, to rename a column within a table, or to add a new column to an existing table.` 😊

**Pull request merged**